app/assets/javascripts/discourse/tests/integration/components/plugin-outlet-test.gjs

```diff
@@ -1058,6 +1059,126 @@ module(
   }
 );
 
+module(
+  "Integration | Component | plugin-outlet | argument currying",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    test("makes arguments available at top level", async function (assert) {
+      extraConnectorComponent(
+        "test-name",
+        <template>
+          <span class="glimmer-test">{{@arg1}} from glimmer</span>
+        </template>
+      );
+
+      await render(
+        <template>
+          <PluginOutlet
+            @name="test-name"
+            @outletArgs={{lazyHash arg1="Hello world"}}
+          />
+        </template>
+      );
+      assert.dom(".glimmer-test").hasText("Hello world from glimmer");
+    });
+
+    test("doesn't completely rerender when arguments change", async function (assert) {
+      const events = [];
+
+      extraConnectorComponent(
+        "test-name",
+        class extends Component {
+          constructor() {
+            super(...arguments);
+            events.push("constructor");
```

Comment:
it doesn't really matter here but this check could also be done with `assert.step()`/`assert.verifySteps`